### PR TITLE
Update `analyzer` dependency

### PR DESCRIPTION
**Pull request by solid-daniilmarchenko**

### CODE REVIEW

**Comment by solid-daniilmarchenko** [comment]
Is this behaviour expected?  
Seems like a bug.

Screenshot 2024-01-19 at 16 23 47


We have two linter warnings around one function: the one right before the 'void' keyword and second near brackets. 

Screenshot 2024-01-19 at 16 22 12

File: solid_lints.iml [hunk 1]
```diff
@@ -9,6 +9,48 @@
       <excludeFolder url="file://$MODULE_DIR$/example/.pub" />
       <excludeFolder url="file://$MODULE_DIR$/example/build" />
       <excludeFolder url="file://$MODULE_DIR$/example/.dart_tool" />
+      <excludeFolder url="file://$MODULE_DIR$/lint_test/.dart_tool" />
```
Comment (yurii-prykhodko-solid) [inline solid_lints.iml:12]: This file probably doesn't belong in the VCS?

**Comment by yurii-prykhodko-solid** [comment]
> Is this behaviour expected?
@solid-daniilmarchenko yeah looks like a bug -- could be stemming from the fact that we're checking both function declaration and function body -- where one contains another, IIRC. Maybe just check function bodies? Give it a try.

**Comment by solid-daniilmarchenko** [comment]
it looks like we really don't need `FunctionDeclaration` but removing it doesn't really fix all of the warning duplications.
i've tried to check other node types (including the `BlockFunctionBody`) that we could use instead of the current ones but it doesnt look to work fine anyway.

File: lib/lints/function_lines_of_code/function_lines_of_code_metric.dart [hunk 2]
```diff
@@ -50,7 +50,6 @@ class FunctionLinesOfCodeMetric
 
     context.registry.addMethodDeclaration(checkNode);
     context.registry.addFunctionExpression(checkNode);
-    context.registry.addFunctionBody(checkNode);
```
Comment (yurii-prykhodko-solid) [inline lib/lints/function_lines_of_code/function_lines_of_code_metric.dart]: Seems to have done the trick with linting all we want with no duplicates

**Review by yurii-prykhodko-solid — APPROVED** [review]
LGTM!

**Pull request merged**